botocore/signers.py

```diff
@@ -130,3 +134,275 @@ def get_auth(self, signing_name, region_name, signature_version=None):
             auth = cls(**kwargs)
             self._cache[key] = auth
             return auth
+
+    def generate_url(self, request_dict, expires_in=3600, region_name=None):
+        """Generates a presigned url
+
+        :type request_dict: dict
+        :param request_dict: The prepared request dictionary returned by
+            ``botocore.awsrequest.prepare_request_dict()``
+
+        :type expires_in: int
+        :param expires_in: The number of seconds the presigned url is valid
+            for. By default it expires in an hour (3600 seconds)
+
+        :type region_name: string
+        :param region_name: The region name to sign the presigned url.
+
+        returns: The presigned url
+        """
+        if region_name is None:
+            region_name = self._region_name
+        query_prefix = '-query'
+        signature_version = self._signature_version
+        if not signature_version.endswith(query_prefix):
+            signature_version += query_prefix
+
+        kwargs = {'signing_name': self._signing_name,
+                  'region_name': region_name,
+                  'signature_version': signature_version,
+                  'expires': expires_in}
+
+        auth = self.get_auth(**kwargs)
+        request = create_request_object(request_dict)
+
+        # Fix s3 host for s3 sigv2 bucket names
+        signature_type = signature_version.split('-', 1)[0]
+        fix_s3_host(request, signature_type, region_name)
+
+        auth.add_auth(request)
+        request.prepare()
+
+        return request.url
+
+    def build_post_form_args(self, request_dict, fields=None, conditions=None,
```

Comment:
I get that these are the names from boto, but I think we should be consistent in our naming, in one case we're using generate, in another we're using build.  I prefer generate.